README.md
[![Build Status](https://travis-ci.org/Aiyoumi-FE/aym-ui.svg?branch=master)](https://travis-ci.org/Aiyoumi-FE/aym-ui)
[![Commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](http://commitizen.github.io/cz-cli/)

## 爱又米专用 VUE 组件库。

`aym-ui` 是基于 Vue 的移动端组件库

### 原 gitHub 文档

[文档地址](https://aiyoumi-fe.github.io/aym-ui/document.html#/)

[示例地址](https://aiyoumi-fe.github.io/aym-ui/examples.html#/)

### 使用说明

#### 安装

```
npm install @fe/aym-ui --save
```

#### 使用

全量使用

```javascript
import aymUI from '@fe/aym-ui'
import '@fe/aym-ui/index.css'
Vue.use(aymUI)
```

按需引入

需要安装插件[babel-plugin-component](https://github.com/ElementUI/babel-plugin-component)

```
npm i babel-plugin-component -D
```

使用方法

```
import { Button } from 'aym-ui'
```

配置`.babelrc`

```javascript
"plugins": [
        "transform-runtime", [
            "component",
            {
                "libraryName": "@fe/aym-ui",
                "libraryDirectory": "lib",
                "style": true
            }
        ]
    ]
```

<!--
按模块引用

npm install babel-plugin-import --save-dev -->

### 开发说明

-   组件源码目录：src/components
-   demo 目录： examples
-   document: document

### 构建说明

-   `npm run dev` 运行 examples,调用 src/index.js 组件出口，组件修改后可以实时预览
-   `npm run build:demo` 构建 demo 和 docs 文件到 dist 目录
-   `npm run build:mui` 构建 js，通过 src/index.js 打包到 libs 目录
-   `npm run build` 包含`npm run build:mui`和`npm run build:modules`两个个命令
-   `npm run deploy:docs` 打包 demo 和 docs 并推送到 github
-   `npm run new xx` 在 examples/pages、document/components/docs 和 src/components 目录下分别加入组件空模版


CHANGELOG.md
## [4.0.1](https://github.com/Aiyoumi-FE/aym-ui/compare/v4.0.0...v4.0.1) (2021-04-15)


### Bug Fixes

* picker scrollEnd repeat ([cf2f54c](https://github.com/Aiyoumi-FE/aym-ui/commit/cf2f54c))


### Features

* modal弹窗maskClick的bug修复 ([585e152](https://github.com/Aiyoumi-FE/aym-ui/commit/585e152))



# [4.0.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v3.0.0-beta.1...v4.0.0) (2019-11-13)


### Bug Fixes

* 移动端兼容性-修改diabled下active有效果问题 ([f37f128](https://github.com/Aiyoumi-FE/aym-ui/commit/f37f128))
* **vue:** 按钮示例间距调整 ([c0a170b](https://github.com/Aiyoumi-FE/aym-ui/commit/c0a170b))
* 优化button ([9c4e327](https://github.com/Aiyoumi-FE/aym-ui/commit/9c4e327))
* **input:** 倒计时销毁 ([9222ef7](https://github.com/Aiyoumi-FE/aym-ui/commit/9222ef7))


### Features

* **button:** 更新button组件 ([84cde79](https://github.com/Aiyoumi-FE/aym-ui/commit/84cde79))
* **cell:** cell 根据4.0规范更新 ([569acf4](https://github.com/Aiyoumi-FE/aym-ui/commit/569acf4))
* **cell,radio,tabbar,marquee:** 组合类组件扁平调用 ([dc731fb](https://github.com/Aiyoumi-FE/aym-ui/commit/dc731fb))
* **color:** update color ([813efed](https://github.com/Aiyoumi-FE/aym-ui/commit/813efed))
* **color,fontsize:** 规范化变量名&全局替换 ([b1900d9](https://github.com/Aiyoumi-FE/aym-ui/commit/b1900d9))
* **demo:** 示例修改 ([70578ac](https://github.com/Aiyoumi-FE/aym-ui/commit/70578ac))
* **input:** 去掉上下结构示例 ([7d986c6](https://github.com/Aiyoumi-FE/aym-ui/commit/7d986c6))
* **popup,modal,actionsheet,piker:** 弹层样式修改 ([9defa95](https://github.com/Aiyoumi-FE/aym-ui/commit/9defa95))
* 合并样式文件 ([ef0c758](https://github.com/Aiyoumi-FE/aym-ui/commit/ef0c758))
* **vue:** 更新tag组件 ([6f89013](https://github.com/Aiyoumi-FE/aym-ui/commit/6f89013))



# [3.0.0-beta.1](https://github.com/Aiyoumi-FE/aym-ui/compare/v3.0.0-beta.0...v3.0.0-beta.1) (2019-06-13)



# [3.0.0-beta.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.3.0...v3.0.0-beta.0) (2019-06-13)


### Bug Fixes

* **cell:** change slot title  tag ([702189a](https://github.com/Aiyoumi-FE/aym-ui/commit/702189a))
* **reset:** img width ([792a712](https://github.com/Aiyoumi-FE/aym-ui/commit/792a712))


### Features

* **cell:** cell unit up ([18fa7aa](https://github.com/Aiyoumi-FE/aym-ui/commit/18fa7aa))
* **cell:** cell组件根据ui规范 更新 ([ef7dbc5](https://github.com/Aiyoumi-FE/aym-ui/commit/ef7dbc5))
* **color:** update base color var ([fd0b910](https://github.com/Aiyoumi-FE/aym-ui/commit/fd0b910))
* **icon:** 图标独立库管理 ([9eaebdb](https://github.com/Aiyoumi-FE/aym-ui/commit/9eaebdb))
* **input:** input组件根据ui规范 更新 ([379bf2e](https://github.com/Aiyoumi-FE/aym-ui/commit/379bf2e))
* **input:** unit update ([838cfd1](https://github.com/Aiyoumi-FE/aym-ui/commit/838cfd1))
* **radio:** radio 更新 ([d8676ef](https://github.com/Aiyoumi-FE/aym-ui/commit/d8676ef))
* **tabbar:** 增加位置属性 ([3edad73](https://github.com/Aiyoumi-FE/aym-ui/commit/3edad73))



# [2.3.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.2.0...v2.3.0) (2019-03-26)


### Bug Fixes

* **group:** 组合组件导出 ([f1b985c](https://github.com/Aiyoumi-FE/aym-ui/commit/f1b985c))
* **progress:** autoplay ([dc96f26](https://github.com/Aiyoumi-FE/aym-ui/commit/dc96f26))



# [2.2.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.1.2...v2.2.0) (2019-03-12)


### Features

* **progress:** 新增半圆进度条组件 ([cdc3108](https://github.com/Aiyoumi-FE/aym-ui/commit/cdc3108))



## [2.1.2](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.1.1...v2.1.2) (2019-03-06)



## [2.1.1](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.1.0...v2.1.1) (2019-03-01)


### Bug Fixes

* **toast:** toast loading style ([1f72a6f](https://github.com/Aiyoumi-FE/aym-ui/commit/1f72a6f))



# [2.1.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.0.0...v2.1.0) (2019-02-25)


### Bug Fixes

* fix no parent ([1ebc241](https://github.com/Aiyoumi-FE/aym-ui/commit/1ebc241))
* **cell:** cell-group cell  组件之间有其他内容时样式展示修复 ([e182d98](https://github.com/Aiyoumi-FE/aym-ui/commit/e182d98))


### Features

* **radio:** add new component ([e6a2bac](https://github.com/Aiyoumi-FE/aym-ui/commit/e6a2bac))
* **sass:** 全局注入变量 ([63345fc](https://github.com/Aiyoumi-FE/aym-ui/commit/63345fc))



# [2.0.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v2.0.0-beta.0...v2.0.0) (2019-01-16)



# [2.0.0-beta.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.4.2...v2.0.0-beta.0) (2019-01-03)


### Features

* **path:** change to releative path,support code ([6290f76](https://github.com/Aiyoumi-FE/aym-ui/commit/6290f76))



## [1.4.2](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.4.1...v1.4.2) (2018-12-21)


### Reverts

* 修正上一个版本错误 ([6922a92](https://github.com/Aiyoumi-FE/aym-ui/commit/6922a92))



## [1.4.1](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.4.0...v1.4.1) (2018-12-21)



# [1.4.0](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.8...v1.4.0) (2018-12-21)


### Features

* **component:** name 前缀可配 ([6742a49](https://github.com/Aiyoumi-FE/aym-ui/commit/6742a49))
* **popup:** 弹层类组件支持maskclick事件 ([f6a784e](https://github.com/Aiyoumi-FE/aym-ui/commit/f6a784e))



## [1.3.8](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.7...v1.3.8) (2018-11-14)


### Bug Fixes

* **mp:** 修复小程序不支持0=》0% ([80a07be](https://github.com/Aiyoumi-FE/aym-ui/commit/80a07be))


### Features

* **auto:** 自动生成release说明推送到git ([aad13e7](https://github.com/Aiyoumi-FE/aym-ui/commit/aad13e7))
* **auto:** changelog ([6617d5a](https://github.com/Aiyoumi-FE/aym-ui/commit/6617d5a))



## [1.3.7](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.6...v1.3.7) (2018-10-29)



## [1.3.6](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.5...v1.3.6) (2018-10-29)



## [1.3.5](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.4...v1.3.5) (2018-10-26)



## [1.3.4](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.3...v1.3.4) (2018-10-26)



## [1.3.3](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.2...v1.3.3) (2018-07-30)



## [1.3.2](https://github.com/Aiyoumi-FE/aym-ui/compare/v1.3.1...v1.3.2) (2018-07-26)




sw.js
importScripts('workbox-sw.prod.v2.1.3.js');

/**
 * DO NOT EDIT THE FILE MANIFEST ENTRY
 *
 * The method precache() does the following:
 * 1. Cache URLs in the manifest to a local cache.
 * 2. When a network request is made for any of these URLs the response
 *    will ALWAYS comes from the cache, NEVER the network.
 * 3. When the service worker changes ONLY assets with a revision change are
 *    updated, old cache entries are left as is.
 *
 * By changing the file manifest manually, your users may end up not receiving
 * new versions of files because the revision hasn't changed.
 *
 * Please use workbox-build or some other tool / approach to generate the file
 * manifest which accounts for changes to local files and update the revision
 * accordingly.
 */
const fileManifest = [
  {
    "url": "document.html",
    "revision": "151b21b16e332b68bcd176b7de21b60d"
  },
  {
    "url": "examples.html",
    "revision": "d2ebaffbcebefb0cfa5b46e36c370277"
  },
  {
    "url": "static/css/document.57c7da827cab98f2f3fa034d471287d6.css",
    "revision": "45fd85870a5c2e081f460a5c0764d38c"
  },
  {
    "url": "static/css/examples.4ec44d30f236aa594ea3cd469777a8f5.css",
    "revision": "8c8e52b1f2a1167e70193175bab06608"
  },
  {
    "url": "static/js/0.8aa5b64.js",
    "revision": "ef47e2db6941d53bbf9415b2d2214908"
  },
  {
    "url": "static/js/1.d0a8021.js",
    "revision": "55848bbd1d47c3d83f861c3aad9707ae"
  },
  {
    "url": "static/js/10.5f861a1.js",
    "revision": "d1b20de22a362bb58aafaf3f6b9b6cb8"
  },
  {
    "url": "static/js/11.3caf64b.js",
    "revision": "2a62670479012d03f46519678556e8ad"
  },
  {
    "url": "static/js/12.013701f.js",
    "revision": "1efe1c508549d663c3507ad6aec97aa1"
  },
  {
    "url": "static/js/13.ec7c371.js",
    "revision": "0911e2ae1456011c062884c4bd8a9f90"
  },
  {
    "url": "static/js/14.eef26c1.js",
    "revision": "77d0aa4a3de8fbdcb5b988db6d70d9b3"
  },
  {
    "url": "static/js/15.a2f8d81.js",
    "revision": "3dca9ef4ed9c7c657d9a24ed6c6f4807"
  },
  {
    "url": "static/js/16.aed2ea2.js",
    "revision": "e3d4eec9d8a1c58a01d99a684e0a6e9a"
  },
  {
    "url": "static/js/17.9bc2f10.js",
    "revision": "c204a19c4e706119a1ac496c90b2ece8"
  },
  {
    "url": "static/js/18.b45f042.js",
    "revision": "2721ee3c3d384c7e6b88e92b61a7bc57"
  },
  {
    "url": "static/js/19.555406f.js",
    "revision": "7bf332b1c626546e34117e2e645dd913"
  },
  {
    "url": "static/js/2.fb90f00.js",
    "revision": "cf3390856e417314deb35b2f6c447ab2"
  },
  {
    "url": "static/js/20.4d60e45.js",
    "revision": "8f6ae6860d5572255229a7691e91bdd5"
  },
  {
    "url": "static/js/3.17f2605.js",
    "revision": "8bc48f8367328c395b94ffb5e42b0140"
  },
  {
    "url": "static/js/4.38fa997.js",
    "revision": "9f086e1f0eb5d69e415ae012cb189740"
  },
  {
    "url": "static/js/5.327eb28.js",
    "revision": "afe1ce8c10e2eed0e8ff7c2a75eeafca"
  },
  {
    "url": "static/js/6.8e1f496.js",
    "revision": "991c131aca29ca41409d3576ab321582"
  },
  {
    "url": "static/js/7.e4191cd.js",
    "revision": "e35235dd3a13b7c34805baa69287f7ef"
  },
  {
    "url": "static/js/8.d86926a.js",
    "revision": "18cf987edb48ee7f6d97e8bd16710543"
  },
  {
    "url": "static/js/9.cd9e7a0.js",
    "revision": "56784f81bdbc4bd0db42b8cfd9ef6b26"
  },
  {
    "url": "static/js/document.786506c.js",
    "revision": "0762cef238bb033bf7dcbae98d8e7b1d"
  },
  {
    "url": "static/js/examples.99d270e.js",
    "revision": "20f47c593c4192f6b155be272443efc5"
  },
  {
    "url": "static/js/manifest.7515c47.js",
    "revision": "52f50bdcbf2ba867b1d4377c274e685a"
  },
  {
    "url": "static/js/vendor.13d250c.js",
    "revision": "264b0bd8d255d127fcd817303525b41b"
  }
];

const workboxSW = new self.WorkboxSW({
  "skipWaiting": true,
  "clientsClaim": true
});
workboxSW.precache(fileManifest);
